### DISCUSSION TITLE
CentOS install

### DISCUSSION THREAD

**Discussion by mortya** [DefectDojo/django-DefectDojo#5786] [General]
My workplace prefers a CentOS-based standalone install rather than a Debian- or Ubuntu-based install or image.  More than a year back, I locally ported setup.bash to get it working under CentOS for a proof of concept install.  Unfortunately, I didn't contribute it back.  Now I'm pulling that project off the backburner, and see that setup.bash has been dropped altogether.  Tentatively, if I got the goDojo installer to work with CentOS and submitted a pull request, is this something that the project would be willing to accept?  I don't want to even consider doing this otherwise.

**Comment by mtesauro** [comment]
So, before I get into about adding CentOS to godojo, there's a couple of things:

- There is an existing document/guide on installing DefectDojo with PostgreSQL using docker-compose  in the community contribs repo - I wrote it so I'm aware of it's existence ;-) You'll find it at https://github.com/DefectDojo/Community-Contribs/tree/master/CentOS-7-deploy  That might just solve your problem.
- Just as an FYI: You are aware that RedHat is drastically changing what CentOS is/will be, right? https://www.theregister.com/2020/12/09/centos_red_hat/ So maybe there's some value in you looking at something like [Rocky Linux](https://rockylinux.org/) or [AlmaLinux](https://almalinux.org/) before too long.  

About adding CentOS support to godojo:

- I'm not at all opposed to adding CentOS support to godojo.  Happy to take PRs against the dev branch of the project at https://github.com/DefectDojo/godojo  If you don't know Go (or don't want to write in Go), no worries.  All I really need is the commands to exec on CentOS vs Debian/Ubuntu Linux.  A PR that just lists the commands would be SUPER useful.
- If you have your port of setup.bash, adding that as an issue to the godojo repo would also be SUPER helpful as I wrote most of setup.bash and can pull the commands I need out of there as well.

The command that are Debian/Ubuntu specific are can be found in the following ways (sorry, I'm about 50% done refactoring how I do the commands to make 'ports' like you asked about easier):

Search the godojo repo for the following to find commands godojo exec's: 
"cmds = " - there's loads of these in the ubuntu.go source file as well as os.go, database.go, targets.go
"cmds: " - there's these in ubuntu.go and database.go
"addCmd" - there's those in ubuntu.go (partial refactor)

If you pull out the Ubuntu specific commands above and give me CentOS ones instead, I can add CentOS as a supported install target in godojo.

HTH
